src/options/project_graphid.rs

```diff
@@ -1,116 +1,95 @@
-use crate::RoverResult;
+use crate::command::init::graph_id::{generate_unique_graph_id, validate_and_check_availability};
+use crate::{RoverError, RoverResult};
 use clap::arg;
 use clap::Parser;
 use dialoguer::Input;
+use rover_client::blocking::StudioClient;
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone, Serialize, Deserialize, Parser, Default)]
+#[derive(Debug, Clone, Serialize, Deserialize, Parser)]
 pub struct GraphIdOpt {
     #[arg(long = "graph-id")]
     pub graph_id: Option<String>,
 }
 
-impl GraphIdOpt {
-    pub fn get_graph_id(&self) -> Option<String> {
-        self.graph_id.clone()
-    }
-
-    pub fn suggest_graph_id(project_name: &str) -> String {
-        format!("{}-graph@current", project_name)
-    }
-
-    pub fn prompt_for_graph_id(suggested_id: &str) -> RoverResult<String> {
-        let graph_id = Input::<String>::new()
-            .with_prompt("Confirm or modify graph ID (start with a letter and use only letters, numbers, and dashes)".to_string())
-            .default(suggested_id.to_string())
-            .validate_with(|input: &String| -> Result<(), &str> {
-                if input.is_empty() {
-                    return Err("Graph ID cannot be empty");
-                }
-                Ok(())
-            })
-            .interact()?;
-
-        Ok(graph_id)
+impl Default for GraphIdOpt {
+    fn default() -> Self {
+        Self { graph_id: None }
     }
+}
 
-    pub fn get_or_prompt_graph_id(&self, project_name: &str) -> RoverResult<String> {
-        if let Some(id) = self.get_graph_id() {
-            // TODO: Validate graph ID format
-            return Ok(id);
+impl GraphIdOpt {
+    fn get_initial_graph_id(&self, project_name: &str) -> String {
+        // If user provides in the command line use that
+        if let Some(ref id) = self.graph_id {
+            id.clone()
+        } else {
+            // Otherwise use suggested default
+            generate_unique_graph_id(project_name)
         }
-
-        let suggested_id = Self::suggest_graph_id(project_name);
-
-        let graph_id = Self::prompt_for_graph_id(&suggested_id)?;
-
-        Ok(graph_id)
     }
-}
-
-// TODO: Add tests for interactive prompts and sad paths
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn test_get_graph_id_with_preset_value() {
-        let instance = GraphIdOpt {
-            graph_id: Some("my-graph".to_string()),
-        };
 
-        let result = instance.get_graph_id();
-        assert_eq!(result, Some("my-graph".to_string()));
-    }
+    pub async fn get_or_prompt_graph_id(
+        &self,
+        client: &StudioClient,
+        project_name: &str,
+    ) -> RoverResult<String> {
+        let initial_id = self.get_initial_graph_id(project_name);
+
+        // If a value is provided, validate it without prompting
+        if self.graph_id.is_some() {
+            validate_and_check_availability(&initial_id, client).await?;
+            return Ok(initial_id);
+        }
```

Comment:
nit: this was a little confusing to read because of the relationship with `self.graph_id` and `self.get_initial_graph_id`.

I would have expected us to validate the `self.graph_id` based on the comment:
```suggestion
        // If a value is provided, validate it without prompting
        if self.graph_id.is_some() {
            validate_and_check_availability(&self.graph_id, client).await?;
            return Ok(self.graph_id);
        }
```

It seems like we could decouple getting the set graph id and generating a unique id.

Reply:
Good point -- I've updated it; let me know if the updated logic clarifies that confusion!